testsuite/perf_utils.py

```diff
@@ -0,0 +1,102 @@
+"""
+This file contains methods that are used in performance testing
+"""
+import os
+from urllib.parse import urlparse, ParseResult
+from importlib import resources
+
+import yaml
+from hyperfoil.factories import HyperfoilFactory, Benchmark
+
+
+def _load_benchmark(filename):
+    """Loads benchmark"""
+    with open(filename, encoding="utf8") as file:
+        benchmark = Benchmark(yaml.load(file, Loader=yaml.Loader))
+    return benchmark
+
+
+def authority(url: str):
+    """Returns hyperfoil authority format of URL <hostname>:<port> from given URL."""
+    parsed_url = urlparse(url)
+    return f"{parsed_url.hostname}:{parsed_url.port}"
+
+
+def prepare_url(url: ParseResult) -> ParseResult:
+    """ Adds port number to url if it is not set"""
+    if not url.port:
+        url_port = 80 if url.scheme == 'http' else 443
+        url = url._replace(netloc=url.hostname + f':{url_port}')
+    return url
+
+
+class HyperfoilUtils:
```

Comment:
You could implement `LifecycleObject` to make it safer. (currently if something in init fails it wont be cleaned up)